A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_ECAMGauge.js

```diff
@@ -0,0 +1,421 @@
+            this.rootSVG.appendChild(this.currentValueText);
+            if (_gaugeDefinition.currentValueBorderWidth > 0) {
+                var borderWidth = this.viewBoxSize.x * _gaugeDefinition.currentValueBorderWidth;
+                var borderHeight = this.currentValueBorderHeight;
+                var borderPosX = textPosX - (borderWidth * 0.95);
+                var borderPosY = textPosY - (borderHeight * 0.5);
+                var currentValueBorder = document.createElementNS(Avionics.SVG.NS, "rect");
+                currentValueBorder.id = "CurrentValueBorder";
+                currentValueBorder.setAttribute("x", borderPosX.toString());
+                currentValueBorder.setAttribute("y", borderPosY.toString());
+                currentValueBorder.setAttribute("width", borderWidth.toString());
+                currentValueBorder.setAttribute("height", borderHeight.toString());
+                this.rootSVG.appendChild(currentValueBorder);
+            }
+            if (this.extraMessageFunction != null) {
+                var extraMessageGroup = document.createElementNS(Avionics.SVG.NS, "g");
+                extraMessageGroup.id = "ExtraMessage";
+                this.extraMessageBorder = document.createElementNS(Avionics.SVG.NS, "rect");
+                this.extraMessageBorder.setAttribute("x", this.extraMessageBorderPosX.toString());
+                this.extraMessageBorder.setAttribute("y", this.extraMessageBorderPosY.toString());
+                this.extraMessageBorder.setAttribute("width", this.extraMessageBorderWidth.toString());
+                this.extraMessageBorder.setAttribute("height", this.extraMessageBorderHeight.toString());
+                this.extraMessageBorder.setAttribute("class", "inactive");
+                extraMessageGroup.appendChild(this.extraMessageBorder);
+                this.extraMessageText = document.createElementNS(Avionics.SVG.NS, "text");
+                this.extraMessageText.setAttribute("x", this.extraMessagePosX.toString());
+                this.extraMessageText.setAttribute("y", this.extraMessagePosY.toString());
+                this.extraMessageText.setAttribute("alignment-baseline", "central");
+                this.extraMessageText.setAttribute("class", "inactive");
+                extraMessageGroup.appendChild(this.extraMessageText);
+                this.rootSVG.appendChild(extraMessageGroup);
+            }
+            this.refreshMainValue(this.minValue, true);
+            if (this.outerIndicatorFunction != null) {
+                this.refreshOuterIndicator(0, true);
+            }
+            if (this.outerDynamicArcFunction != null) {
+                this.refreshOuterDynamicArc(0, 0);
+            }
+            this.refreshActiveState();
+        }
+
+        //accepts two more parameters to set custom ID for dynamic markers
+        addGraduation(_value, _showInnerMarker, _text = "", _showOuterMarker = false, _setid = false, _idName = "") {
```

Comment:
updated addgraduation to now accept ID to set custom ID for the marker and use the same later with the dynamic function.